presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -309,6 +339,7 @@ private Expression unwrapCast(ComparisonExpression expression)
                         return TRUE_LITERAL;
                     case LESS_THAN:
                     case LESS_THAN_OR_EQUAL:
+                        // this condition will never be true for PARTIAL injective types
```

Comment:
This comment looks like a general rule for partial injective types, but it's not.
Now, that the out-of-bound cases are supported, it is not obvious that the condition is false. Actually, it is false because such cases would be handled / excluded until this point.
(see the comment on code structure)